pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTScriptBlock.java

```diff
@@ -7,13 +7,17 @@
  */
 public class ModelASTScriptBlock extends ModelASTStep {
     public ModelASTScriptBlock(Object sourceLocation) {
+        this(sourceLocation, "script");
+    }
+
+    protected ModelASTScriptBlock(Object sourceLocation, String name) {
```

Comment:
:bug: I'm not a huge fan of overloading `ModelASTScriptBlock` - I'd rather keep it as just for `script` blocks. I'd say go with making a new class that has this as the main constructor, with `ModelASTWhen` and `ModelASTScriptBlock` both extending that new class.